lang/c/src/value-json.c

```diff
@@ -352,26 +354,40 @@ avro_value_to_json_t(const avro_value_t *value)
 			union_schema = avro_value_get_schema(value);
 			branch_schema =
 			    avro_schema_union_branch(union_schema, disc);
+
+			namespace = avro_schema_namespace(branch_schema);
 			branch_name = avro_schema_type_name(branch_schema);
+			if (namespace != NULL) {
+				full_name_buf = json_sprintf("%s.%s", namespace, branch_name);
+				if (full_name_buf == NULL) {
+					avro_set_error("Cannot allocate full name union");
+					return NULL;
+				}
+				branch_name = json_string_value(full_name_buf);
+			}
 
 			json_t  *result = json_object();
 			if (result == NULL) {
 				avro_set_error("Cannot allocate JSON union");
+				json_decref(full_name_buf);
```

Comment:
`full_name_buf` could be NULL if there is no namespace. Maybe we need to check before calling json_decref() ?!
Same issues below.

Reply:
`json_decref` [supports](https://github.com/akheron/jansson/blob/master/src/jansson.h#L131) dereferencing `NULL`.
So I do not think this is a problem.